### PR TITLE
Ajouter le filtre de `nature juridique`

### LINKED ISSUE #106: Filtre par nature juridique

### LINKED ISSUE #162: API de Recherche : filtre nature juridique

### PR DESCRIPTION
**Pull request by HAEKADI**
closes #106 
closes #162

### CODE REVIEW

**Review by geoffreyaldebert — APPROVED** [review]
code structure is very clean 👏

**Comment by geoffreyaldebert** [comment]
but please lint this code

**Comment by HAEKADI** [comment]
@geoffreyaldebert It's the same lint issue (push vs pull request), it will disappear once I merge...

File: aio/aio-proxy/aio_proxy/doc/open-api.yml [hunk 1]
```diff
@@ -264,6 +264,15 @@ paths:
           example: "0022004T"
           schema:
             type: string
+        - name: nature_juridique
+          in: query
+          description: >-
```
Comment (XavierJp) [inline aio/aio-proxy/aio_proxy/doc/open-api.yml:269]: Maybe a link to a valid list ?

File: aio/aio-proxy/aio_proxy/parsers/nature_juridique.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        list_nature_juridique_clean if valid.
+
+    Raises:
+        ValueError: if one of the values in list_nature_juridique_clean is not valid.
+    """
+    if list_nature_juridique_clean is None:
+        return None
+    for nature_juridique in list_nature_juridique_clean:
+        if len(nature_juridique) != 4 or not nature_juridique.isdigit():
+            raise ValueError("Au moins une nature juridique est non valide.")
```
Comment (XavierJp) [inline aio/aio-proxy/aio_proxy/parsers/nature_juridique.py:22]: Pourquoi pas retourner la liste des codes valides comme pour les départements ?

**Review by XavierJp — APPROVED** [review]
C'est une tuerie

File: aio/aio-proxy/aio_proxy/doc/open-api.yml (hunk 1 above)
Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/doc/open-api.yml:269]: I couldn't find a full list of values. https://www.insee.fr/fr/information/2028129

File: aio/aio-proxy/aio_proxy/parsers/nature_juridique.py (hunk 2 above)
Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/parsers/nature_juridique.py:22]: From what I understand in this page: there is a 9*37*260 possible values which seems like too many values to return don't you think ?

Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/parsers/nature_juridique.py:22]: I can add a more detailed check with list of values.

Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/parsers/nature_juridique.py]: Done !

**Pull request merged**